librz/crypto/crypto.c

```diff
@@ -98,22 +74,47 @@ RZ_API int rz_crypto_del(RzCrypto *cry, RzCryptoPlugin *h) {
 	return true;
 }
 
-RZ_API struct rz_crypto_t *rz_crypto_new(void) {
+RZ_API RzCrypto *rz_crypto_new(void) {
 	RzCrypto *cry = RZ_NEW0(RzCrypto);
-	return rz_crypto_init(cry, true);
-}
+	if (!cry) {
+		rz_warn_if_reached();
+		goto rz_crypto_new_bad;
+	}
+
+	cry->output_size = RZ_CRYPTO_OUTPUT_SIZE;
+	cry->output = malloc(RZ_CRYPTO_OUTPUT_SIZE);
+	if (!cry->output) {
+		rz_warn_if_reached();
```

Comment:
I am not sure we need `rz_warn_if_reached()` here since it is mostly exists for programmatic errors rather than runtime, like `rz_return_if_fail()` macro. cc @ret2libc

Reply:
i'll change them with RZ_ERR_LOG